src/main/java/pl/edu/icm/pl/mxrdr/extension/workflow/XdsAdjustResultStep.java

```diff
@@ -0,0 +1,245 @@
+    }
+
+    // -------------------- LOGIC --------------------
+
+    @Override
+    protected WorkflowStepResult.Source runInternal(WorkflowExecutionContext workflowExecutionContext, Path workDir) throws Exception {
+        try {
+            FileContentReplacer.of(XDS_INP, workDir)
+                    .add(PatternAndReplacement.of(
+                            JOB_PARAM_REGEX, JOB_PARAM_REPLACEMENT + sanitizeJobList(jobs)))
+                    .add(shouldProcessResolutionRange
+                            ? PatternAndReplacement.of(RESOLUTION_PARAM_REGEX,
+                            RESOLUTION_PARAM_REPLACEMENT + ResolutionParameterExtractor.of(workDir).extract())
+                            : PatternAndReplacement.NONE)
+                    .replace();
+            return Success.successWith(output -> Collections.emptyMap());
+        } catch (RuntimeException re) {
+            logger.warn("Exception during step execution: ", re);
+            return ooutput -> new Failure(re.getMessage());
+        }
+    }
+
+    @Override
+    public WorkflowStepResult resume(WorkflowExecutionContext context, Map<String, String> internalData, String externalData) {
+        throw new UnsupportedOperationException("This step does not pause");
+    }
+
+    @Override
+    public void rollback(WorkflowExecutionContext workflowExecutionContext, Failure reason) { }
+
+    // -------------------- PRIVATE --------------------
+
+    private String sanitizeJobList(String jobList) {
+        return jobList.replaceAll(DELIMITERS_REGEX, " ")
+                .replaceAll("\\s+", " ");
+    }
+
+    // -------------------- INNER CLASSES --------------------
+
+    static class FileContentReplacer {
+        private String fileName;
+        private Path workDir;
+        private Set<PatternAndReplacement> actions = new HashSet<>();
+
+        // -------------------- CONSTRUCTORS --------------------
+
+        private FileContentReplacer(String fileName, Path workDir) {
+            this.fileName = fileName;
+            this.workDir = workDir;
+        }
+
+        // -------------------- LOGIC --------------------
+
+        public static FileContentReplacer of(String fileName, Path workDir) {
+            return new FileContentReplacer(fileName, workDir);
+        }
+
+        public FileContentReplacer add(PatternAndReplacement action) {
+            if (action != PatternAndReplacement.NONE) {
+                actions.add(action);
+            }
+            return this;
+        }
+
+        public void replace() {
+            File file = workDir.resolve(fileName).toFile();
+            File modifiedFile = workDir.resolve(fileName + RandomStringUtils.randomAlphanumeric(8)).toFile();
+            try (BufferedReader reader = new BufferedReader(new FileReader(file));
+                 BufferedWriter writer = new BufferedWriter(new FileWriter(modifiedFile))) {
+                readReplaceAndWrite(reader, writer);
```

Comment:
Consider closing reader and writer after operation

Reply:
AFAIK the closing of these resources should happen automatically, as they are within try-with-resources construct and both implement the AutoCloseable interface.